backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsync.java

```diff
@@ -0,0 +1,197 @@
+                extractedData.getState().getValue(),
+                extractedData.getZipCode().getValue(),
+                null);
+
+        String country =
+            extractedData.getCountry().getValue() == null
+                ? "USA"
+                : extractedData.getCountry().getValue();
+
+        if (_personService.isDuplicatePatient(
+            extractedData.getFirstName().getValue(),
+            extractedData.getLastName().getValue(),
+            parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+            currentOrganization,
+            assignedFacility)) {
+          continue;
+        }
+
+        // create new person with current organization, then add to new patients list
+        Person newPatient =
+            new Person(
+                currentOrganization,
+                assignedFacility.orElse(null),
+                null, // lookupid
+                extractedData.getFirstName().getValue(),
+                extractedData.getMiddleName().getValue(),
+                extractedData.getLastName().getValue(),
+                extractedData.getSuffix().getValue(),
+                parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+                address,
+                country,
+                parsePersonRole(extractedData.getRole().getValue(), false),
+                List.of(extractedData.getEmail().getValue()),
+                convertRaceToDatabaseValue(extractedData.getRace().getValue()),
+                convertEthnicityToDatabaseValue(extractedData.getEthnicity().getValue()),
+                null, // tribalAffiliation
+                convertSexToDatabaseValue(extractedData.getBiologicalSex().getValue()),
+                parseYesNo(extractedData.getResidentCongregateSetting().getValue()),
+                parseYesNo(extractedData.getEmployedInHealthcare().getValue()),
+                null, // preferredLanguage
+                null // testResultDeliveryPreference
+                );
+
+        if (!patientsList.contains(newPatient)) {
+          // collect phone numbers and associate them with the patient
+          // then add to phone numbers list and set primary phone, if exists
+          List<PhoneNumber> newPhoneNumbers =
+              _personService.assignPhoneNumbersToPatient(
+                  newPatient,
+                  List.of(
+                      new PhoneNumber(
+                          parsePhoneType(extractedData.getPhoneNumberType().getValue()),
+                          extractedData.getPhoneNumber().getValue())));
+          phoneNumbersList.addAll(newPhoneNumbers);
+          newPhoneNumbers.stream().findFirst().ifPresent(newPatient::setPrimaryPhone);
+
+          patientsList.add(newPatient);
+        }
+      } catch (IllegalArgumentException e) {
+        sendErrorEmail(uploaderEmail, simplereportUrl, currentOrganization);
+        String errorMessage = "Error uploading patient roster";
+        log.error(
+            errorMessage
```

Comment:
This block of logging code is duplicated in the next try/catch for database saves - could you extract it out into a `logFailure` method or similar?

Reply:
✅